example/simple/src/main.c

```diff
@@ -360,49 +355,61 @@ int main(int argc, char *argv[])
 
 	/* Initialize completion semaphore */
 	ret = sem_init(&context.sem, 0, 0);
-	if (ret)
-		error("Unable to initialize completion semaphore\n");
+	if (ret) {
+		printf("Error: Unable to initialize completion semaphore\n");
+		goto sem_err;
+	}
 	context.error = &ret;
 
 	/* Initialize OCF context */
-	if (ctx_init(&ctx))
-		error("Unable to initialize context\n");
+	if (ctx_init(&ctx)) {
+		printf("Error: Unable to initialize context\n");
+		goto ctx_err;
+	}
 
 	/* Start cache */
-	if (initialize_cache(ctx, &cache1))
-		error("Unable to start cache\n");
+	if (initialize_cache(ctx, &cache1)) {
+		printf("Error: Unable to start cache\n");
+		goto cache_err;
+	}
 
 	/* Add core */
-	if (initialize_core(cache1, &core1))
-		error("Unable to add core\n");
+	if (initialize_core(cache1, &core1)) {
+		printf("Error: Unable to add core\n");
+		goto core_err;
+	}
 
 	/* Do some actual io operations */
 	perform_workload(core1);
 
 	/* Remove core from cache */
 	ocf_mngt_cache_remove_core(core1, remove_core_complete, &context);
 	sem_wait(&context.sem);
-	if (ret)
-		error("Unable to remove core\n");
+	if (ret) {
+		printf("Error: Unable to remove core\n");
+		goto core_err;
+	}
 
 	/* Stop cache */
 	ocf_mngt_cache_stop(cache1, simple_complete, &context);
 	sem_wait(&context.sem);
-	if (ret)
-		error("Unable to stop cache\n");
+	if (ret) {
```

Comment:
'ret' is not set for this check

Reply:
The address of `ret` is assigned to `context.error` in line 362 so the value would be updated by `simple_complete()`